cmd/update_selector.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/syntasso/kratix/api/v1alpha1"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+var updateSelectorCmd = &cobra.Command{
+	Use:   "selector KEY:VALUE",
+	Short: "Command to update destination selectors",
+	Long:  "Command to update destination selectors",
+	Example: `  # add a new destination selector
+  kratix update selector env:dev
+  # removes an existing destination selector
+  kratix update selector zone-
+`,
+	RunE: UpdateSelector,
+	Args: cobra.ExactArgs(1),
+}
+
+func init() {
+	updateCmd.AddCommand(updateSelectorCmd)
+	updateSelectorCmd.Flags().StringVarP(&dir, "dir", "d", ".", "Directory to read Promise from")
+}
+
+func UpdateSelector(cmd *cobra.Command, args []string) error {
+	var promise v1alpha1.Promise
+	if promise, err = getPromise(filepath.Join(dir, "promise.yaml")); err != nil {
+		return fmt.Errorf("failed to find promise.yaml in directory: %v", err)
+	}
+
+	parsed := strings.Split(args[0], ":")
+	if len(parsed) != 2 {
```

Comment:
thoughts on reversing the flow?

```golang
if len(parsed) == 2 {
   // do the else block
} else {
  // do the if block
}
```

for some reason my brain finds it easier to grok -- feel free to ignore

Reply:
Updated to a switch statement. Does that read better?

Reply:
i think it does, although i'm not sure we will ever have something other than the case 2 😆 

my previous suggestion was based on `if x != y { A } else { B }` reads something like 
> if x is not equal to y
>   do A; 
> else, x is not not equal to Y
>   do B

with the positive condition, it reads something like:

> if x is equal to y
>   do A; 
> else, x is not to Y
>   do B

I'm fine with the switch though 👍🏽